File: 2022/AdventOfCode2022/Input.cs
using System;
using System.IO;

namespace AdventOfCode2022
{
    public class Input
    {
        public string Raw { get; init; }
        public string[] Lines { get; init; }

        public Input(string file)
        {
            Raw = File.ReadAllText(file);
            Lines = Raw.Split(new string[] { "\n", "\r\n" }, StringSplitOptions.None);
        }
    }
}

File: 2023/input.py
class Input:
    def __init__(self, path: str):
        with open(path, "r") as file:
            self._raw = file.read()
            file.seek(0)
            self._lines = [line.rstrip() for line in file.readlines()]

    @property
    def raw(self) -> str:
        return self._raw
    
    @property
    def lines(self) -> list[str]:
        return self._lines
